src/Server/HTTP/HTTPServerRequest.cpp

```diff
@@ -63,11 +63,9 @@ HTTPServerRequest::HTTPServerRequest(HTTPContextPtr context, HTTPServerResponse
     }
     else if (getMethod() != HTTPRequest::HTTP_GET && getMethod() != HTTPRequest::HTTP_HEAD && getMethod() != HTTPRequest::HTTP_DELETE)
     {
-        /// That check for has_body may be false-negative in rare cases, but it's okay
-        bool has_body = in->hasPendingData();
         stream = std::move(in);
-        if (!startsWith(getContentType(), "multipart/form-data") && has_body)
-            LOG_WARNING(&Poco::Logger::get("HTTPServerRequest"), "Got an HTTP request with no content length "
+        if (!startsWith(getContentType(), "multipart/form-data"))
+            LOG_WARNING(LogFrequencyLimiter(&Poco::Logger::get("HTTPServerRequest"), 10), "Got an HTTP request with no content length "
```

Comment:
But `LogFrequencyLimiterIml` does not filter messages without a format string. Not sure how this message is converted into Poco::Message `message.getFormatString()`

Reply:
This message has a "format string" with no parameters, so `message.getFormatString()` will return the message itself. Logging without a format string looks like this:
https://github.com/ClickHouse/ClickHouse/blob/4a5e4274f03f4d320b9e3d0c6315b38084255c8e/src/Dictionaries/CassandraHelpers.cpp#L57-L61